File: lib/models/shoe_model.dart
class Shoe {
  final int id;
  final String name;
  final String imageUrl;
  final String description;
  final double price;
  final String color;

  Shoe(this.id, this.name, this.imageUrl, this.description, this.price,
      this.color);

  Shoe.fromJson(Map<String, dynamic> json)
      : id = json['id'] as int,
        name = json['name'] as String,
        imageUrl = json['image'] as String,
        description = json['description'] as String,
        price = json['price'] as double,
        color = json['color'] as String;

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'image': imageUrl,
        'description': description,
        'price': price,
        'color': color
      };

  @override
  String toString() {
    return """
      Shoe ID: $id
      name: $name
      description: $description
      price: $price
      ---------------------------
    """;
  }
}

File: src/models/shoe.model.ts
import Item from './item.model';

class Shoe implements Item {
  id: number;
  image: string;
  name: string;
  description: string;
  price: number;
  color: string;

  constructor(
    id: number,
    imageUrl: string,
    name: string,
    description: string,
    price: number,
    color: string,
  ) {
    this.id = id;
    this.name = name;
    this.image = imageUrl;
    this.description = description;
    this.price = price;
    this.color = color;
  }

  static fromJson(props: ShoeDTO): Shoe {
    return new Shoe(props.id, props.image, props.name, props.description, props.price, props.color);
  }
}

interface ShoeDTO {
  id: number;
  image: string;
  name: string;
  description: string;
  price: number;
  color: string;
}

export default Shoe;
